packages/tools/examples/changeColormap/index.ts

```diff
@@ -0,0 +1,247 @@
+
+// Define unique ids for the volumes
+const volumeLoaderScheme = 'cornerstoneStreamingImageVolume'; // Loader id which defines which volume loader to use
+const ctVolumeName = 'CT_VOLUME_ID'; // Id of the volume less loader prefix
+const ctVolumeId = `${volumeLoaderScheme}:${ctVolumeName}`; // VolumeId with loader id + volume id
+
+// Define a unique id for the volume
+const ptVolumeName = 'PT_VOLUME_ID';
+const ptVolumeId = `${volumeLoaderScheme}:${ptVolumeName}`;
+
+// ======== Set up page ======== //
+setTitleAndDescription(
+  'Volume Viewport API With Multiple Volumes',
+  'Demonstrates how to interact with a Volume viewport when using fusion.'
+);
+
+const content = document.getElementById('content');
+const element = document.createElement('div');
+element.id = 'cornerstone-element';
+element.style.width = '500px';
+element.style.height = '500px';
+
+content.appendChild(element);
+// ============================= //
+
+// TODO -> Maybe some of these implementations should be pushed down to some API
+
+// Buttons
+
+let fused = false;
+let opacity = 0;
+addButtonToToolbar({
+  title: 'toggle PET',
+  onClick: async () => {
+    // Get the rendering engine
+    const renderingEngine = getRenderingEngine(renderingEngineId);
+
+    // Get the volume viewport
+    const viewport = <Types.IVolumeViewport>(
+      renderingEngine.getViewport(viewportId)
+    );
+    if (fused) {
+      viewport.removeVolumeActors([ptVolumeId], true);
+
+      fused = false;
+    } else {
+      await viewport.addVolumes(
+        [
+          {
+            volumeId: ptVolumeId,
+          },
+        ],
+        true
+      );
+
+      viewport.setProperties(
+        {
+          colormap: {
+            name: 'hsv',
+            opacity: opacity,
+          },
+          voiRange: {
+            upper: 5,
+            lower: 0,
+          },
+        },
+        ptVolumeId
+      );
+      fused = true;
+    }
+  },
+});
+
+addButtonToToolbar({
+  title: 'Change Colomap',
+  onClick: () => {
+    // Get the rendering engine
+    const renderingEngine = getRenderingEngine(renderingEngineId);
+
+    // Get the volume viewport
+    const viewport = <Types.IVolumeViewport>(
+      renderingEngine.getViewport(viewportId)
+    );
+    const randomIndex = Math.floor(
+      Math.random() * vtkColorMaps.rgbPresetNames.length
+    );
+
+    const colormapName = vtkColorMaps.rgbPresetNames[randomIndex];
+
+    viewport.setProperties(
+      {
+        colormap: {
+          name: colormapName,
+        },
+      },
+      ptVolumeId
+    );
+    viewport.render();
+  },
+});
+addSliderToToolbar({
+  title: 'opacity',
+  step: 0.001,
+  range: [0, 255],
+  defaultValue: 0,
+  onSelectedValueChange: (value) => {
+    const valueAsNumber = Number(value) / 255;
+
+    opacity = valueAsNumber;
+    // Get the rendering engine
+    const renderingEngine = getRenderingEngine(renderingEngineId);
+
+    // Get the volume viewport
+    const viewport = <Types.IVolumeViewport>(
```

Comment:
you can hoist the viewport outside these functions so that you have them once

like

```js
let viewport

/// 

// enable 


viewport = getRenderingEngine()

```